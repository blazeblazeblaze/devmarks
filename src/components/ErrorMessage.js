import React from "react";
import PropTypes from "prop-types";

const propTypes = {
  onClick: PropTypes.func
};

const ErrorMessage = ({ onClick }) => (
  <section className="section">
    <div className="container">
      <article className="message is-large is-danger">
        <div className="message-body">
          Oops! An unexpected error occured.
          <p>
            {onClick ? (
              <a
                className="button is-rounded"
                href=""
                onClick={e => {
                  e.preventDefault();
                  onClick();
                }}
              >
                <strong>Try again</strong>
              </a>
            ) : (
              ""
            )}
          </p>
        </div>
      </article>
    </div>
  </section>
);

ErrorMessage.propTypes = propTypes;

export default ErrorMessage;
