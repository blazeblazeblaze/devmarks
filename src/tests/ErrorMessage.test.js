import React from "react";
import ErrorMessage from "../components/ErrorMessage";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<ErrorMessage />", () => {
  describe("when onClick handled is provided", () => {
    const onClickHandler = jest.fn();
    const tree = shallow(<ErrorMessage onClick={onClickHandler} />);

    it("renders retry button", () => {
      expect(tree.find(".button").exists()).toBe(true);
    });

    it("matches snapshot", () => {
      expect(toJson(tree)).toMatchSnapshot();
    });

    it("calls handler when button is clicked", () => {
      tree.find(".button").simulate("click", { preventDefault: jest.fn() });
      expect(onClickHandler).toHaveBeenCalled();
    });
  });

  describe("when onClick handled is not provided", () => {
    it("renders retry button", () => {
      const tree = shallow(<ErrorMessage />);
      expect(tree.find(".button").exists()).toBe(false);
    });

    it("matches snapshot", () => {
      const tree = shallow(<ErrorMessage />);
      expect(toJson(tree)).toMatchSnapshot();
    });
  });
});
