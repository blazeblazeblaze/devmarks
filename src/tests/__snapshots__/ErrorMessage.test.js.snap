// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> when onClick handled is not provided matches snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <article
      className="message is-large is-danger"
    >
      <div
        className="message-body"
      >
        Oops! An unexpected error occured.
        <p />
      </div>
    </article>
  </div>
</section>
`;

exports[`<ErrorMessage /> when onClick handled is provided matches snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <article
      className="message is-large is-danger"
    >
      <div
        className="message-body"
      >
        Oops! An unexpected error occured.
        <p>
          <a
            className="button is-rounded"
            href=""
            onClick={[Function]}
          >
            <strong>
              Try again
            </strong>
          </a>
        </p>
      </div>
    </article>
  </div>
</section>
`;
